### PR TITLE
Pop-ups for wallmounts

### PR DESCRIPTION
**Pull request by Time-Green**
## About The Pull Request

https://github.com/user-attachments/assets/05ecc958-37a9-4684-8c64-cf8fbd62a156

(emissives are fixed!!!)

Adds pop-ups for east, west and north facing wallmounts! Whenever you hold shift, they appear on certain commonly used wallmounts. 

The overlays are unclickable, you'll still need to actually click the wallmount. I could probably make it count as an attack (by making image overlays have different click behaviour to not make it automatically examine? it's probably possible but ughhh), but then it could genuinely obstruct objects, which is a something I'm not ready for.

## Why It's Good For The Game

I genuinely struggled on the testmerge with finding and differentiating wallmounts, and while it can be improved with spritework and by getting used to it, I don't see the harm in having this as well. It's not intended as a crutch to start implementing tons of north facing posters, it's just here as a visual aid. 

It's also on the shift key since shift is already the "i'm curious" / "i want more information" key. You can rebind it or unbind it to get rid of it

## Idea's
Maybe make the wallmount pop-ups not only go up? It can be a little cluttered, but then I'd be putting it in hallways (it's not clientside it's plane fuckery so I can't always place it in The Black. 

Also maybe outlines? Or colors? I added a hologram effect but it was kinda meh

Thanks to Watermelon and Lemon for helping me with planes and images stuff!

### CODE REVIEW

**Comment by Time-Green** [comment]
Changed the arrow sprite? 
image
Will probably play with it a bit more but I'm out of time for now

**Comment by Time-Green** [comment]
Probably good enough?
image
If not it will irritate a spriter into doing a way better job than me

**Comment by loganuk** [comment]
Looks really cool

It'll be good for the non urgent game play items like posters and flavour items.

**Comment by Time-Green** [comment]
I'm really struggling on getting the emissives to not be emissive. 
image
image

I can't modify the overlays directly (or by having the list send through signals), and flags like KEEP_TOGETHER all make exceptions for planes

**Comment by LemonInTheDark** [comment]
> I can't modify the overlays directly ....

Whyyyyyy not?

File: code/datums/components/examine_balloon.dm [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	atom_parent.update_icon(UPDATE_OVERLAYS)
+
+/datum/component/examine_balloon/proc/on_update_icon(atom/movable/parent, updates)
+	SIGNAL_HANDLER
+
+	if(!(updates & UPDATE_OVERLAYS))
+		return
+
+	parent.overlays.Cut() //Clear all previous overlays, so we don't infinitely stack with our own overlays
+	parent.managed_overlays.Cut()
```
Comment (LemonInTheDark) [inline code/datums/components/examine_balloon.dm]: this isn't valid, more things then you might try to use the overlays list. either here or in `on_updated_icon` you should just `cut_overlay(hologram) // same thing for the arrow`

File: code/datums/components/examine_balloon.dm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	if(!(updates & UPDATE_OVERLAYS))
+		return
+
+	// Generally south facing directions are already obvious, so we dont add a hologram (south is the default exception dont shoot me)
+	if(!(parent.dir & draw_in_dirs))
+		return
+
+	// Make a copy of the wallmount and force it south
+	var/mutable_appearance/hologram = make_mutable_appearance_directional(new /mutable_appearance (parent), SOUTH)
```
Comment (LemonInTheDark) [inline code/datums/components/examine_balloon.dm]:
```suggestion
	var/mutable_appearance/hologram = make_mutable_appearance_directional(new /mutable_appearance(parent), SOUTH)
```

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by Time-Green** [comment]
@LemonInTheDark IT WORKS!!!!

File: code/datums/components/examine_balloon.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/datums/components/examine_balloon.dm]: not resolved

Comment (LemonInTheDark) [reply code/datums/components/examine_balloon.dm]: also you both should not be and have no reason to touch managed_overlays (unless I have forgotten something)

**Comment by LemonInTheDark** [comment]
> @LemonInTheDark IT WORKS!!!!

POGGED

File: code/datums/components/examine_balloon.dm [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	hologram.pixel_w = 0
+	hologram.pixel_x = 0
+	hologram.pixel_y = pixel_y_offset
+	hologram.pixel_z = 0
+
+	// We need to set all our overlay's planes to the wallmount balloons plane, or we get stuff like emissives sticking through the lighting plane
+	var/list/new_overlays = list()
+	for(var/mutable_appearance/immutable_appearance as anything in parent.overlays)
+		var/mutable_appearance/actually_mutable_appearance = new(immutable_appearance)
+		if(PLANE_TO_TRUE(actually_mutable_appearance.plane) == EMISSIVE_PLANE)
```
Comment (LemonInTheDark) [inline code/datums/components/examine_balloon.dm]: is there any plane that WOULD be acceptable or should we only accept things with FLOAT_PLANE or a plane value that matches the parent (that we can set to FLOAT_PLANE)

File: code/datums/components/examine_balloon.dm (hunk 1 above)
Comment (Time-Green) [reply code/datums/components/examine_balloon.dm]: cut_overlays() does not clear the managed overlays, meaning that on next icon update it compares the new overlays with managed overlays, and if they're equal it doesn't update the icon. In effect this means wallmounts don't have overlays if they call update_icons without having a different appearance

also I see now that I completely misread what you said. I did try that but it didn't work but I tried again and now it works so ???? 🎉🎉🎉

File: code/datums/components/examine_balloon.dm (hunk 3 above)
Comment (Time-Green) [reply code/datums/components/examine_balloon.dm]: I don't have enough plane knowledge, but I ran it with only accepting FLOAT_PLANE and everything looks good
image

**Comment by Time-Green** [comment]
@LemonInTheDark Reviews processed!

**Comment by LemonInTheDark** [comment]
image
image
Not 100% on board w action but fuck it we ball

**Comment by Time-Green** [comment]
Yeah I get that. It’s partially sending a wrong message that vague sprites are okay, which isn’t intended but kinda inevitable. Ideally this is in place till every wallmount is clearly distinct, and then maybe restricted to only posters or something

**Pull request merged**